cmd/veneur-emit/main_test.go

```diff
@@ -256,6 +282,170 @@ func TestSendSpan(t *testing.T) {
 	}
 }
 
+func TestBadSendSpan(t *testing.T) {
+	testFlag := make(map[string]flag.Value)
+	dataWritten = []byte{}
+	conn := &badConn{}
+	span, _ := createMetrics(testFlag, "test.metric", "tag1:value1")
+	err := sendSpan(conn, span)
+	assert.NotNil(t, err, "Did not return err!")
+}
+
+func TestBuildEventPacketError(t *testing.T) {
+	testFlag := make(map[string]flag.Value)
+	_, err := buildEventPacket(testFlag)
+	assert.NotNil(t, err, "Did not catch error.")
+
+	testFlag["e_title"] = newValue("myBadTitle")
+	_, err = buildEventPacket(testFlag)
+	assert.NotNil(t, err, "Did not catch error. (title, no text)")
+
+	testFlag = make(map[string]flag.Value)
+	testFlag["e_text"] = newValue("myBadText")
+	_, err = buildEventPacket(testFlag)
+	assert.NotNil(t, err, "Did not catch error. (text, no title)")
+}
+
+func TestBuildEventPacket(t *testing.T) {
+	myTitle := "myTitle"
+	myText := "myText"
+
+	testFlag := make(map[string]flag.Value)
+
+	t.Run("simple", func(t *testing.T) {
+		testFlag["e_title"] = newValue(myTitle)
+		testFlag["e_text"] = newValue(myText)
+
+		pkt, err := buildEventPacket(testFlag)
+		assert.Nil(t, err, "Returned non-nil error.")
+
+		expected := fmt.Sprintf("_e{%d,%d}:%s|%s", len(myTitle), len(myText), myTitle, myText)
+		assert.Equal(t, pkt.String(), expected, "Bad event packet.")
+	})
+
+	t.Run("basic", func(t *testing.T) {
+		testFlag["e_title"] = newValue("An exception occurred")
+		testFlag["e_text"] = newValue("Cannot parse CSV file from 10.0.0.17")
+		testFlag["e_alert_type"] = newValue("warning")
+		tags := "tag1:value1"
+		testFlag["e_event_tags"] = newValue(tags)
+
+		pkt, err := buildEventPacket(testFlag)
+		assert.Nil(t, err, "Returned non-nil error.")
+
+		suffix := fmt.Sprintf("#%s", tags)
+		assert.True(t, strings.HasSuffix(pkt.String(), suffix), "Tags not at end of event packet.")
+
+		assert.True(t, strings.HasPrefix(pkt.String(), "_e"))
+	})
+
+	// TODO: check this test
```

Comment:
Can't you verify that when you pass in something like `"foo\nbar"` to `buildEventPacket` it comes out as `"foo\\nbar"`?